src/RemoteConfigContext.tsx

```diff
@@ -80,6 +91,8 @@ const RemoteConfigContextProvider: React.FC = ({children}) => {
     }
   }, [isLoadingAppState, newBuildSincePreviousLaunch]);
 
+  useInterval(fetchConfig, RETRY_INTERVAL_MS, undefined, !fetchError);
```

Comment:
I know there are some restriction on how often and how many times is allowed to request data from the firebase servers, because it might generate some throttles as described here: https://firebase.google.com/docs/remote-config/get-started?&platform=ios#throttling and some other restrictions for `fetchAndActivate` as mention in the documentation:

```If the time elapsed since the last fetch from the Firebase Remote Config backend is more than the default minimum fetch interval, configs are fetched from the backend.```

The documentation also mentions:

```
Don't rely on network connectivity to obtain Remote Config values. Do set in-app default parameter values so that your app always behaves as expected. You can periodically keep app and Remote Config backend default values in sync using [downloaded template defaults](https://firebase.google.com/docs/remote-config/templates#download_template_defaults).
```
They mention this Template defaults.

So, I am not sure if all this applies for this case because we do not know what `fetchAndActivate` register and / or knows there was a network issue, but I wanted to bring it.

This solution seems correct for me, just a bit aware of side cases.

Reply:
At least the retry mechanism is a "soft refresh", so it shouldn't end up being throttled since it will find the values in the cache first if the time since last fetch from backend hasn't surpassed the minimum fetch interval.